zipline/data/us_equity_pricing.py

```diff
@@ -528,6 +528,16 @@ def _spot_col(self, colname):
     def get_last_traded_dt(self, asset, day):
         volumes = self._spot_col('volume')
         search_day = day
+
+        if search_day >= asset.end_date:
+            # move to the last day before the asset end date
+            try:
+                search_day = self._calendar[
+                    self._calendar.get_loc(asset.end_date) - 1
```

Comment:
I believe this can also be accomplished by `asset.end_date - self._calendar.freq`.
(Though not 100% if this calendar object has a `freq`).